packages/common/src/types/transactionEvent.ts

```diff
@@ -0,0 +1,352 @@
+    from?: Address;
+}
+
+export interface TransferredWithScheduleEvent {
+    tag: TransactionEventTag.TransferredWithSchedule;
+    to: Base58String;
+    from?: Base58String;
+    amount: ReleaseSchedule[];
+}
+
+export interface MemoEvent {
+    tag: TransactionEventTag.TransferMemo;
+    memo: HexString;
+}
+
+export interface AccountCreatedEvent {
+    tag: TransactionEventTag.AccountCreated;
+    account: Base58String;
+}
+
+export interface AmountAddedByDecryptionEvent {
+    tag: TransactionEventTag.AmountAddedByDecryption;
+    account?: Base58String;
+    amount: bigint;
+}
+
+export interface EncryptedSelfAmountAddedEvent {
+    tag: TransactionEventTag.EncryptedSelfAmountAdded;
+    account: Base58String;
+    amount: bigint;
+    newAmount: string;
+}
+
+export interface EncryptedAmountsRemovedEvent {
+    tag: TransactionEventTag.EncryptedAmountsRemoved;
+    account?: Base58String;
+    inputAmount: HexString;
+    newAmount: HexString;
+    upToindex: number;
+}
+
+export interface NewEncryptedAmountEvent {
+    tag: TransactionEventTag.NewEncryptedAmount;
+    account: Base58String;
+    newIndex: number;
+    encryptedAmount: HexString;
+}
+
+export interface CredentialDeployedEvent {
+    tag: TransactionEventTag.CredentialDeployed;
+    regid: HexString;
+    account: Base58String;
+}
+
+export interface CredentialKeysUpdatedEvent {
+    tag: TransactionEventTag.CredentialKeysUpdated;
+    credId: HexString;
+}
+
+export interface CredentialsUpdatedEvent {
+    tag: TransactionEventTag.CredentialsUpdated;
+    account?: Base58String;
+    newCredIds: HexString[];
+    removedCredIDs: HexString[];
+    newThreshold: number;
+}
+
+// Delegation Events
+
+export interface DelegatorEvent {
+    tag:
+        | TransactionEventTag.DelegationAdded
+        | TransactionEventTag.DelegationRemoved;
+    delegatorId: number;
+    account?: Base58String;
+}
+
+export interface DelegationSetDelegationTargetEvent {
+    tag: TransactionEventTag.DelegationSetDelegationTarget;
+    delegatorId: number;
+    account?: Base58String;
+    delegationTarget: EventDelegationTarget;
+}
+
+export interface DelegationSetRestakeEarningsEvent {
+    tag: TransactionEventTag.DelegationSetRestakeEarnings;
+    delegatorId: number;
+    account?: Base58String;
+    restakeEarnings: boolean;
+}
+
+export interface DelegationStakeChangedEvent {
+    tag:
+        | TransactionEventTag.DelegationStakeDecreased
+        | TransactionEventTag.DelegationStakeIncreased;
+    delegatorId: number;
+    account?: Base58String;
+    newStake: bigint;
+}
+
+// Baker Events
+
+export interface BakerAddedEvent {
+    tag: TransactionEventTag.BakerAdded;
+    bakerId: number;
+    account: string;
+    signKey: string;
+    electionKey: string;
+    aggregationKey: string;
+    stake: bigint;
+    restakeEarnings: boolean;
+}
+
+export interface BakerRemovedEvent {
+    tag: TransactionEventTag.BakerRemoved;
+    bakerId: number;
+    account?: Base58String;
```

Comment:
What is the meaning of this optional account address?

Reply:
It is present for v1, but for v2 it is not present. I'll change it to be added in v2.